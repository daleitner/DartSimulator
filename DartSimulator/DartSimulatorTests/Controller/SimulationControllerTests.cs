using ApprovalTests;
using ApprovalTests.Reporters;
using DartBot;
using DartBot.Player;
using DartSimulator.Controller;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Moq;

namespace DartSimulatorTests.Controller
{
	[TestClass]
	[UseReporter(typeof(DiffReporter))]
	public class SimulationControllerTests
	{
		private Mock<IPlayerService> playerMock;

		[TestInitialize]
		public void Setup()
		{
			playerMock = new Mock<IPlayerService>();
		}

		[TestMethod]
		public void WhenStartSimulation_ThenResultShouldHaveTenLegs()
		{
			var legs = 10;
			playerMock.Setup(x => x.PlayLeg()).Returns(new Leg());
			var controller = new SimulationController(playerMock.Object);
			var result = controller.StartSimulation(legs, 100, 100, false);
			Approvals.Verify("Amount of Legs: " + result.Legs.Count);
		}

		[TestMethod]
		public void WhenInitializeRoundCounts_ThenVerifyRoundCounts()
		{
			var controller = new SimulationController(playerMock.Object);
			Approvals.VerifyAll(controller.InitializeRoundCounts(), "RoundCounts");
		}
	}
}
