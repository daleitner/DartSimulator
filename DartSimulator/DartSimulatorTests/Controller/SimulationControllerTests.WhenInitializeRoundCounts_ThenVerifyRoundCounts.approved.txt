RoundCounts[0] = Round: 3, Height: 0, Count: 0
RoundCounts[1] = Round: 4, Height: 0, Count: 0
RoundCounts[2] = Round: 5, Height: 0, Count: 0
RoundCounts[3] = Round: 6, Height: 0, Count: 0
RoundCounts[4] = Round: 7, Height: 0, Count: 0
RoundCounts[5] = Round: 8, Height: 0, Count: 0
RoundCounts[6] = Round: 9, Height: 0, Count: 0
RoundCounts[7] = Round: 10, Height: 0, Count: 0
RoundCounts[8] = Round: 11, Height: 0, Count: 0
RoundCounts[9] = Round: 12, Height: 0, Count: 0
RoundCounts[10] = Round: 13, Height: 0, Count: 0
RoundCounts[11] = Round: 14, Height: 0, Count: 0
RoundCounts[12] = Round: 15, Height: 0, Count: 0
RoundCounts[13] = Round: >15, Height: 0, Count: 0
